src/world.rs

```diff
@@ -0,0 +1,150 @@
+    /// Utility function to test a single path against the defined regexp field and returns a parsed WorldMap if it matches.
+    /// Returns none if the filename does not match the pattern.
+    pub fn capture_path(&self, path: &Path) -> Result<WorldMap, Error> {
+        let re = Regex::new(&self.regexp).unwrap();
+        let captures = re
+            .captures(path.to_str().unwrap())
+            .ok_or(Error::CapturesNotFound)?;
+
+        let x = captures
+            .get(1)
+            .ok_or(Error::CapturesNotFound)?
+            .as_str()
+            .parse::<i32>()
+            .unwrap();
+        let y = captures
+            .get(2)
+            .ok_or(Error::CapturesNotFound)?
+            .as_str()
+            .parse::<i32>()
+            .unwrap();
+
+        // Calculate x and y positions based on the multiplier and offset.
+        let x = x
+            .checked_mul(self.multiplier_x as i32)
+            .ok_or(Error::InvalidPropertyValue {
+                description: "multiplierX causes overflow".to_string(),
+            })?
+            .checked_add(self.offset_x)
+            .ok_or(Error::InvalidPropertyValue {
+                description: "offsetX causes overflow".to_string(),
+            })?;
+
+        let y = y
+            .checked_mul(self.multiplier_y as i32)
+            .ok_or(Error::InvalidPropertyValue {
+                description: "multiplierY causes overflow".to_string(),
+            })?
+            .checked_add(self.offset_y)
+            .ok_or(Error::InvalidPropertyValue {
+                description: "offsetY causes overflow".to_string(),
+            })?;
+
+        Ok(WorldMap {
+            filename: path.to_str().unwrap().to_owned(),
+            x,
+            y,
+            width: None,
+            height: None,
+        })
+    }
+
+    /// Utility function to test a list of paths against the defined regexp field.
+    /// Returns a parsed list of WorldMaps from any matched filenames.
+    pub fn capture_paths(&self, paths: Vec<PathBuf>) -> Result<Vec<WorldMap>, Error> {
+        paths
+            .iter()
+            .map(|path| self.capture_path(path.as_path()))
+            .collect::<Result<Vec<_>, _>>()
+    }
+}
+
+pub(crate) fn parse_world(
+    world_path: &Path,
+    reader: &mut impl ResourceReader,
+) -> Result<World, Error> {
+    let mut path = reader
+        .read_from(&world_path)
+        .map_err(|err| Error::ResourceLoadingError {
+            path: world_path.to_owned(),
+            err: Box::new(err),
+        })?;
+
+    let mut world_string = String::new();
+    path.read_to_string(&mut world_string)
+        .map_err(|err| Error::ResourceLoadingError {
+            path: world_path.to_owned(),
+            err: Box::new(err),
+        })?;
+
+    let world: World = match serde_json::from_str(&world_string) {
+        Ok(world) => world,
+        Err(err) => {
+            return Err(Error::JsonDecodingError(err));
+        }
+    };
+
+    Ok(world)
```

Comment:
I guess this could be simplified to something like (untested):

```suggestion
    serde_json::from_str(&world_string)
        .map_err(|err| Error::JsonDecodingError(err)))
```

Reply:
Good suggestion. Change in latest commit.